be/src/storage/lake/delta_writer.cpp

```diff
@@ -285,13 +279,18 @@ Status DeltaWriterImpl::finish() {
     op_write->mutable_rowset()->set_data_size(_tablet_writer->data_size());
     op_write->mutable_rowset()->set_overlapped(op_write->rowset().segments_size() > 1);
     // handle partial update
-    if (_rowset_txn_meta != nullptr && _partial_update_tablet_schema != nullptr) {
-        op_write->mutable_txn_meta()->CopyFrom(*_rowset_txn_meta);
+    RowsetTxnMetaPB* rowset_txn_meta = _tablet_writer->rowset_txn_meta();
+    if (rowset_txn_meta != nullptr && _partial_update_tablet_schema != nullptr) {
```

Comment:
Question: is it possible to have `rowset_txn_meta != nullptr && _partial_update_tablet_schema == nullptr` or `rowset_txn_meta == nullptr && _partial_update_tablet_schema != nullptr`

Reply:
`rowset_txn_meta` can't be nullptr, if it is primary table. I will use CHECK to make sure this. And _partial_update_tablet_schema can be nullptr when it is normal update, or not nullptr when it is partial update.